src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -72,9 +73,7 @@ private async Task RenderComponentCore(HttpContext context)
         // Matches MVC's MemoryPoolHttpResponseStreamWriterFactory.DefaultBufferSize
         var defaultBufferSize = 16 * 1024;
         await using var writer = new HttpResponseStreamWriter(context.Response.Body, Encoding.UTF8, defaultBufferSize, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
```

Comment:
Aren't we buffering twice? Once here, and once in the BufferedTextWriter?

Reply:
Yes, that's a good point. I retained this usage of `HttpResponseStreamWriter` from the earlier implementation but it might be that we could simply omit it and have `TextChunkListBuilder` able to write directly to `Response.Body`.

I'll benchmark it both ways and go with whatever appears to be more performant. That sound OK?

Reply:
I just saw it and I think we would remove one buffering layer, which is generally good.

Reply:
OK, I looked into this and tried replacing `HttpResponseStreamWriter` with:

 1. `StreamWriter`. This works but is marginally slower than `HttpResponseStreamWriter`. Perhaps the default buffer size in `HttpResponseStreamWriter` is just better tuned to these cases. They are doing mostly the same things.
 2. No string encoding buffer at all and writing directly to the `Stream`. This was much more complicated, since it's necessary to then replicate a lot of what `StreamWriter` or `HttpResponseStreamWriter` do to encode the different types of things we might be writing. And doing it in a simplistic way resulted in much lower perf than the above techniques - I suspect the optimal way might turn out to be the same as copying the internals from `StreamWriter` or `HttpResponseStreamWriter`.

So I think we'll stick with `HttpResponseStreamWriter` as in practice it appears to give the best results, and is the most well-tested code path in this scenario. It was worth looking into though, so thanks for the suggestion!